scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java

```diff
@@ -0,0 +1,399 @@
+package org.cloudfoundry.autoscaler.scheduler.service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.cloudfoundry.autoscaler.scheduler.dao.ScheduleDao;
+import org.cloudfoundry.autoscaler.scheduler.entity.ScheduleEntity;
+import org.cloudfoundry.autoscaler.scheduler.rest.model.ApplicationScalingSchedules;
+import org.cloudfoundry.autoscaler.scheduler.util.DataValidationHelper;
+import org.cloudfoundry.autoscaler.scheduler.util.DateHelper;
+import org.cloudfoundry.autoscaler.scheduler.util.SpecificDateScheduleDateTime;
+import org.cloudfoundry.autoscaler.scheduler.util.ScheduleTypeEnum;
+import org.cloudfoundry.autoscaler.scheduler.util.error.DatabaseValidationException;
+import org.cloudfoundry.autoscaler.scheduler.util.error.SchedulerInternalException;
+import org.cloudfoundry.autoscaler.scheduler.util.error.ValidationErrorResult;
+import org.quartz.SchedulerException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Service class to persist the schedule entity in the database and create
+ * scheduled job.
+ * 
+ * 
+ *
+ */
+@Service
+public class ScheduleManager {
+
+	@Autowired
+	private ScheduleDao scheduleDao;
+	@Autowired
+	private ScheduleJobManager scalingJobManager;
+	@Autowired
+	private ValidationErrorResult validationErrorResult;
+
+	private String scheduleBeingProcessed; // Specific date/Recurring
+	private Logger logger = LogManager.getLogger(this.getClass());
+
+	/**
+	 * Calls dao and fetch all the schedules for the specified application id.
+	 * 
+	 * @param appId
+	 * @return
+	 * @throws Exception
+	 */
+	public ApplicationScalingSchedules getAllSchedules(String appId) {
+		logger.info("Get All schedules for application: " + appId);
+		
+		List<ScheduleEntity> allScheduleEntitiesForApp = null;
+		ApplicationScalingSchedules applicationScalingSchedules = null;
+		try {
+			allScheduleEntitiesForApp = scheduleDao.findAllSchedulesByAppId(appId);
+		} catch (DatabaseValidationException dve) {
+
+			validationErrorResult.addErrorForDatabaseValidationException(dve, "schedule.database.error.get.failed",
+					"app_id=" + appId);
+			throw new SchedulerInternalException("Database error", dve);
+		}
+
+		if (allScheduleEntitiesForApp != null) {
+			applicationScalingSchedules = populateScheduleModel(allScheduleEntitiesForApp);
```

Comment:
I understand the catch part. Just trying to make the code cleaner to understand.